google_takeout_parser/parse_json.py

```diff
@@ -125,12 +125,12 @@ def _parse_app_installs(p: Path) -> Iterator[Res[PlayStoreAppInstall]]:
     for japp in json_data:
         try:
             yield PlayStoreAppInstall(
-                title=japp["install"]["doc"]["title"],
-                deviceName=japp["install"]["deviceAttribute"].get("deviceDisplayName"),
-                deviceCarrier=japp["install"]["deviceAttribute"]["carrier"],
-                deviceManufacturer=japp["install"]["deviceAttribute"]["manufacturer"],
-                lastUpdateTime=parse_json_utc_date(japp["install"]["lastUpdateTime"]),
-                firstInstallationTime=parse_json_utc_date(japp['install']['firstInstallationTime']),
+                title=japp.get("install", {}).get("doc", {}).get("title"),
+                deviceName=japp.get("install", {}).get("deviceAttribute", {}).get("deviceDisplayName"),
+                deviceCarrier=japp.get("install", {}).get("deviceAttribute", {}).get("carrier"),
+                deviceManufacturer=japp.get("install", {}).get("deviceAttribute", {}).get("manufacturer"),
+                lastUpdateTime=parse_json_utc_date(japp.get("install", {}).get("lastUpdateTime")),
```

Comment:
this is fine as a pattern, but `parse_json_utc_date` expects a string, and if the key is missing it'll pass `None` to `parse_json_utc_date` and just fail and crash. So, should probably refactor this upwards and if they key is missing we should just fail to parse and yield a RuntimeError. If we cant find a timestamp that is pretty much a breaking change, thats sort of one of the most core parts of the data we want